src/devices/GoPiGo3/GoPiGo3.cs

```diff
@@ -0,0 +1,900 @@
+        /// <param name="dps">The speed limit in degrees per second, with 0 being no limit</param>
+        public void SetMotorLimits(MotorPort port, byte powerPercent = 0, int dps = 0)
+        {
+            byte[] outArray = { SpiAddress, (byte)SpiMessageType.setMotorLimits, (byte)port, powerPercent, (byte)((dps >> 8) & 0xFF), (byte)(dps & 0xFF) };
+            var ret = SpiTransferArray(outArray);
+        }
+
+        /// <summary>
+        /// Read a motor status
+        /// </summary>
+        /// <param name="port">The Motor port to use, can be MotorLeft or MotorRight</param>
+        /// <returns>MotorStatus containing the status of the motor</returns>
+        public MotorStatus GetMotorStatus(MotorPort port)
+        {
+            MotorStatus motorStatus = new MotorStatus();
+            SpiMessageType message_type = (port == MotorPort.MotorRight) ? SpiMessageType.GetMotorStatusRight : SpiMessageType.GetMotorStatusLeft;
+            byte[] outArray = { SpiAddress, (byte)message_type, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+            var reply = SpiTransferArray(outArray);
+            if (reply[3] == 0xA5)
+            {
+                motorStatus.Speed = reply[5];
+                if ((motorStatus.Speed & 0x80) > 0)
+                    motorStatus.Speed = -motorStatus.Speed;
+                motorStatus.Encoder = (int)(((reply[6] << 24) | (reply[7] << 16) | (reply[8] << 8) | reply[9]) / MotorTicksPerDegree);
+                motorStatus.Dps = ((reply[10] << 8) | reply[11]);
+                if ((motorStatus.Dps & 0x8000) > 0)
+                    motorStatus.Dps = motorStatus.Dps - 0x10000;
+                motorStatus.Flags = (MotorStatusFlags)reply[4];
+            }
+            else
+            {
+                throw new IOException($"{nameof(GetMotorStatus)} error: no SPI response");
+            }
+
+            return motorStatus;
+        }
+
+        /// <summary>
+        /// Offset a motor encoder
+        /// </summary>
+        /// <param name="port">The Motor port to use, can be MotorLeft and/or MotorRight</param>
+        /// <param name="positionOffset">The encoder offset. Zero the encoder by offsetting it by the current position</param>
+        public void OffsetMotorEncoder(MotorPort port, int positionOffset)
+        {
+            positionOffset = (int)(positionOffset * MotorTicksPerDegree);
+            byte[] outArray = new byte[] { SpiAddress, (byte)SpiMessageType.OffsetMotorEncoder, (byte)port, (byte)((positionOffset >> 24) & 0xFF), (byte)((positionOffset >> 16) & 0xFF), (byte)((positionOffset >> 8) & 0xFF), (byte)(positionOffset & 0xFF) };
+            SpiTransferArray(outArray);
+        }
+
+        /// <summary>
+        /// Read a motor encoder in degrees
+        /// </summary>
+        /// <param name="port">The Motor port to use, can be MotorLeft or MotorRight</param>
+        /// <returns>Returns the encoder position in degrees</returns>
+        public int GetMotorEncoder(MotorPort port)
```

Comment:
should this be GetMotorPosition considering you convert encoder ticks to degrees?

Reply:
I would leave it like this as I provide a way to convert to degree. You may want to just use the encoder as a raw data to control another element directly rather than having to convert.